Runtime error, nil dereference in cluster-ctl:main.formatResponse

Running kubernetes-ipfs test 0012-random_kill.yml I found this runtime error while pinning a new hash with cluster-ctl.  The short explanation is that cluster-ctl's main's formatResponse function can get a nil http.Response under bad conditions which causes a nil dereference on the first line of the function.  Note, as this is a pin add operation the offending call to the function is made on line 259.

Should we enforce an invariant that formatResponse never gets called with a nil http.Response and error check before calling, or handle this case as the first line of formatResponse?

The whole golang stack trace:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x5571b35fc399]

goroutine 1 [running]:
main.formatResponse(0xc420012a00, 0x0)
	/go/src/github.com/ipfs/ipfs-cluster/ipfs-cluster-ctl/main.go:482 +0x39
main.main.func6(0xc420012a00, 0x0, 0xc420012a00)
	/go/src/github.com/ipfs/ipfs-cluster/ipfs-cluster-ctl/main.go:259 +0x222
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.HandleAction(0x5571b3b25ac0, 0x5571b3b9c468, 0xc420012a00, 0xc42005a600, 0x0)
	/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/app.go:485 +0xd4
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.Command.Run(0x5571b35ff844, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x5571b360334f, 0x11, 0x0, ...)
	/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/command.go:193 +0xa9b
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.(*App).RunAsSubcommand(0xc42007f1e0, 0xc420012640, 0x0, 0x0)
	/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/app.go:374 +0xa7d
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.Command.startApp(0x5571b35ff892, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/command.go:280 +0x870
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.Command.Run(0x5571b35ff892, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/command.go:79 +0x13ee
gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli.(*App).Run(0xc42007eea0, 0xc4200100c0, 0x4, 0x4, 0x0, 0x0)
	/go/src/gx/ipfs/Qmc1AtgBdoUHP8oYSqU81NRYdzohmF45t5XNwVMvhCxsBA/cli/app.go:250 +0x75a
main.main()
	/go/src/github.com/ipfs/ipfs-cluster/ipfs-cluster-ctl/main.go:429 +0x1b64
```

The offending k8s-ipfs test step:
```
### Running step pin files while potentially being shutdown on nodes [4]
$ ipfs-cluster-ctl pin add ${HASH[%i]} > /tmp/out.txt; if [[ $(cat /tmp/out.txt) = *${HASH[%i]}* ]] && [[ $(cat /tmp/out.txt) != *error* ]] && [[ $(cat /tmp/out.txt | wc -l) = 3 ]]; then echo 'yes'; else echo 'no'; cat /tmp/out.txt; fi
Running parallel on 1 nodes on iteration 1.
```

--- a/ipfs-cluster-ctl/main.go
+++ b/ipfs-cluster-ctl/main.go
@@ -474,7 +474,7 @@ func request(method, path string, body io.Reader, args ...string) *http.Response
 
 	client := &http.Client{Transport: defaultTransport}
 	resp, err := client.Do(r)
-
+	checkErr(fmt.Sprintf("performing request to %s", defaultHost), err)
 	return resp
 }
 

